### PR TITLE
Adds nil handling to alert filtering

### PR DESCRIPTION
**Pull request by ErinLMoore**
Also compares the datetimes correctly

### CODE REVIEW

File: test/trip_updates/trip_updates_pub_sub_test.exs [hunk 1]
```diff
@@ -184,5 +184,27 @@ defmodule TripUpdates.TripUpdatesPubSubTest do
 
       assert TripUpdatesPubSub.is_it_fresh?(stale_tripupdate, current_time) == false
     end
+
+    test "you broke it" do
```
Comment (arkadyan) [inline test/trip_updates/trip_updates_pub_sub_test.exs:188]:
```suggestion
    test "handles nil arrival times" do
```
suggestion (non-blocking): This made me laugh, but maybe could be more descriptive?

File: lib/trip_updates/trip_updates_pub_sub.ex [hunk 2]
```diff
@@ -108,10 +108,18 @@ defmodule TripUpdates.TripUpdatesPubSub do
...
+    else
+      _ -> false
+    end
   end
 
   defp last_stop_arrival(trip_update) do
-    trip_update.stop_time_update |> Enum.map(& &1.arrival_time) |> Enum.max(DateTime)
+    trip_update.stop_time_update
+    |> Enum.map(& &1.arrival_time)
+    |> Enum.filter(&(!is_nil(&1)))
```
Comment (arkadyan) [inline lib/trip_updates/trip_updates_pub_sub.ex:122]: suggestion (non-blocking): There's also `Enum.reject` if you wanted to avoid the `!`.

**Review by arkadyan — APPROVED** [review]
👍

File: test/trip_updates/trip_updates_pub_sub_test.exs (hunk 1 above)
Comment (ErinLMoore) [reply test/trip_updates/trip_updates_pub_sub_test.exs:188]: Oh noooo I meant to change that >_< sorry

**Pull request merged**